src/realm/cluster.cpp

```diff
@@ -1653,24 +1653,68 @@ ref_type Cluster::typed_write(ref_type ref, _impl::ArrayWriterBase& out, const T
             else if (col_type == col_type_Mixed) {
                 const auto sz = leaf.size();
                 REALM_ASSERT(sz == 6);
-                // temporary disable mixed. in order to re-enable them in a separate PR
+
+                /*
+                 In order to know how Mixed stores things, we need to take in consideration this enum
```

Comment:
Better to say we need to take into account the use of the following sub-arrays (and not the enum)